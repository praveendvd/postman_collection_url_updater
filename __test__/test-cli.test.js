const exec = require('shelljs').exec,
  SDK = require('postman-collection'),
  FS = require('fs');

describe('Validate index.js', () => {
  describe('Validate cli output messages', () => {
    it('Should show help if arguments are missing', async () => {
      const commandResponse = exec('node "index.js"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
    });

    it('Should show help if -c argument is missing', async () => {
      const commandResponse = exec('node "index.js" -c "test/collection/collection.json" -w "{{baseURL}}/{{path}}" -s "new_collection.json"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
      expect(commandResponse.stdout).toMatchSnapshot();
    });

    it('Should show help if -r argument is missing', async () => {
      const commandResponse = exec('node "index.js" -c "test/collection/collection.json" -w "{{baseURL}}/{{path}}" -s "new_collection.json"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
      expect(commandResponse.stdout).toMatchSnapshot();
    });

    it('Should show help if -w argument is missing', async () => {
      const commandResponse = exec('node "index.js" -c "test/collection/collection.json" -r "{{baseURL}}/{{path}}" -s "new_collection.json"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
      expect(commandResponse.stdout).toMatchSnapshot();
    });

    it('Should show file not found error if collection doesnt exists', async () => {
      const commandResponse = exec('node "index.js" -c "test/collection/collection.json" -r "https://localhost:23456/api/v1/{{path}}" -w "{{baseURL}}/{{path}}" -s "new_collection.json"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
      expect(commandResponse.stdout).toMatchSnapshot();
    });

    it('Should show file saved message correctly with default path', async () => {
      const commandResponse = exec('node "index.js" -c "__test__/collection/collection.json" -r "https://localhost:23456/api/v1/{{path}}" -w "{{baseURL}}/{{path}}"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
      expect(commandResponse.stdout).toMatchSnapshot();
    });

    it('Should show file saved message correctly with custom path', async () => {
      const commandResponse = exec('node "index.js" -c "__test__/collection/collection.json" -r "https://localhost:23456/api/v1/{{path}}" -w "{{baseURL}}/{{path}}" -s "__test__/collection/output/new.json"', { silent: true });
      expect(commandResponse.stderr).toMatchSnapshot();
      expect(commandResponse.stdout).toMatchSnapshot();
    });
  })

  describe('Validate collection changes', () => {
    let collectionPath = '__test__/collection/collection.json',
      outputCollectionPath = '__test__/collection/output/new.json';

    it('validate no changes happens when there is no match', async () => {
      sourceCollection = new SDK.Collection(JSON.parse(FS.readFileSync(collectionPath).toString()));
      const commandResponse = exec(`node "index.js" -c "${collectionPath}" -r "https://localhost:23456/api/v1/{{path}}" -w "{{baseURL}}/{{path}}" -s "new_collection.json" -s "${outputCollectionPath}"`, { silent: true });
      outputCollection = new SDK.Collection(JSON.parse(FS.readFileSync(outputCollectionPath).toString()));
      expect(sourceCollection).toStrictEqual(outputCollection)
    });

  })
})


