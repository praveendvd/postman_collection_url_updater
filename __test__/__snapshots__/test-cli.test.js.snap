// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate index.js system test Validate cli output messages Should show file not found error if collection doesnt exists 1`] = `
"ENOENT: no such file or directory, open 'test/collection/collection.json'
"
`;

exports[`Validate index.js system test Validate cli output messages Should show file not found error if collection doesnt exists 2`] = `""`;

exports[`Validate index.js system test Validate cli output messages Should show help if -c argument is missing 1`] = `
"Usage: -c \\"root/collection_name.json\\" -r \\"{{Baseurl}}/path1/path2\\" -w
\\"{{Baseurl}}/{{path}}\\" -s \\"root/new_collection_name.json\\"

Options:
      --help              Show help                                    [boolean]
      --version           Show version number                          [boolean]
  -c, --collection_path   Path to Collection file            [string] [required]
  -r, --replace_url_part  Replaces only the matching part of the URL
                                                             [string] [required]
  -w, --with_url_part     Replaces the matching part of the URL with provided
                          value                              [string] [required]
  -s, --save_as           path to save new collection
                           [string] [default: \\"root/new_<collection_name>.json\\"]

Missing required argument: c
"
`;

exports[`Validate index.js system test Validate cli output messages Should show help if -c argument is missing 2`] = `""`;

exports[`Validate index.js system test Validate cli output messages Should show help if -r argument is missing 1`] = `
"Usage: -c \\"root/collection_name.json\\" -r \\"{{Baseurl}}/path1/path2\\" -w
\\"{{Baseurl}}/{{path}}\\" -s \\"root/new_collection_name.json\\"

Options:
      --help              Show help                                    [boolean]
      --version           Show version number                          [boolean]
  -c, --collection_path   Path to Collection file            [string] [required]
  -r, --replace_url_part  Replaces only the matching part of the URL
                                                             [string] [required]
  -w, --with_url_part     Replaces the matching part of the URL with provided
                          value                              [string] [required]
  -s, --save_as           path to save new collection
                           [string] [default: \\"root/new_<collection_name>.json\\"]

Missing required argument: r
"
`;

exports[`Validate index.js system test Validate cli output messages Should show help if -r argument is missing 2`] = `""`;

exports[`Validate index.js system test Validate cli output messages Should show help if -w argument is missing 1`] = `
"Usage: -c \\"root/collection_name.json\\" -r \\"{{Baseurl}}/path1/path2\\" -w
\\"{{Baseurl}}/{{path}}\\" -s \\"root/new_collection_name.json\\"

Options:
      --help              Show help                                    [boolean]
      --version           Show version number                          [boolean]
  -c, --collection_path   Path to Collection file            [string] [required]
  -r, --replace_url_part  Replaces only the matching part of the URL
                                                             [string] [required]
  -w, --with_url_part     Replaces the matching part of the URL with provided
                          value                              [string] [required]
  -s, --save_as           path to save new collection
                           [string] [default: \\"root/new_<collection_name>.json\\"]

Missing required argument: w
"
`;

exports[`Validate index.js system test Validate cli output messages Should show help if -w argument is missing 2`] = `""`;

exports[`Validate index.js system test Validate cli output messages Should show help if arguments are missing 1`] = `
"Usage: -c \\"root/collection_name.json\\" -r \\"{{Baseurl}}/path1/path2\\" -w
\\"{{Baseurl}}/{{path}}\\" -s \\"root/new_collection_name.json\\"

Options:
      --help              Show help                                    [boolean]
      --version           Show version number                          [boolean]
  -c, --collection_path   Path to Collection file            [string] [required]
  -r, --replace_url_part  Replaces only the matching part of the URL
                                                             [string] [required]
  -w, --with_url_part     Replaces the matching part of the URL with provided
                          value                              [string] [required]
  -s, --save_as           path to save new collection
                           [string] [default: \\"root/new_<collection_name>.json\\"]

Missing required arguments: c, r, w
"
`;
